### PR TITLE
drop Python 2.6 and 3.3 support

### LINKED ISSUE #679: With setuptools dropping python2.6, do we finally have enough precedent to drop it ourselves?
http://setuptools.readthedocs.io/en/latest/history.html#v37-0-0

### PR DESCRIPTION
**Pull request by gaborbernat**
Resolves #679. 

Took the opportunity to also update our documentation, so we no longer refer to ``Python <2.7`` or ``Python <=3.3`` in there.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -59,21 +51,20 @@ def main():
         author_email='holger@merlinux.eu',
         packages=['tox'],
         entry_points={'console_scripts': 'tox=tox:cmdline\ntox-quickstart=tox._quickstart:main'},
+        python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*',
```
Comment (nicoddemus) [inline setup.py]: 👍 😄

File: changelog/679.feature.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+drop Pyhton ``2.6`` and ``3.3`` support: ``setuptools`` dropped supporting these, and as we depend on it we'll follow
+up with doing the same (use ``tox <= 2.9.1`` if you still need this support) - by @gaborbernat
```
Comment (nicoddemus) [inline changelog/679.feature.rst]: You can mention that a recent enough `pip` (>=9) should automatically install < 2.9 on Python 2.6 and Python 3.3.

File: doc/example/nose.rst [hunk 3]
```diff
@@ -25,7 +25,7 @@ and the following ``tox.ini`` content:
 
 you can invoke ``tox`` in the directory where your ``tox.ini`` resides.
 ``tox`` will sdist-package your project create two virtualenv environments
-with the ``python2.6`` and ``python2.5`` interpreters, respectively, and will
+with the ``python3.6`` and ``python2.5`` interpreters, respectively, and will
```
Comment (nicoddemus) [inline doc/example/nose.rst]: Probably worth updating `python2.5` to a more modern version as well

File: doc/index.rst [hunk 4]
```diff
@@ -68,7 +68,7 @@ Current features
 * uses pip_ and setuptools_ by default.  Support for configuring the installer command
   through :confval:`install_command=ARGV`.
 
-* **cross-Python compatible**: CPython-2.6, 2.7, 3.2 and higher,
+* **cross-Python compatible**: CPython-2.7, 3.3 and higher,
```
Comment (nicoddemus) [inline doc/index.rst]: *3.4 and higher

File: setup.py [hunk 5]
```diff
@@ -32,21 +32,13 @@ def get_long_description():
...
 def main():
-    version = sys.version_info[:2]
     virtualenv_open = ['virtualenv>=1.11.2']
-    virtualenv_capped = ['virtualenv>=1.11.2,<14']
     install_requires = ['py>=1.4.17', 'pluggy>=0.3.0,<1.0', 'six']
     extras_require = {}
     if has_environment_marker_support():
-        extras_require[':python_version=="2.6"'] = ['argparse']
-        extras_require[':python_version=="3.2"'] = virtualenv_capped
         extras_require[':python_version!="3.2"'] = virtualenv_open
```
Comment (nicoddemus) [inline setup.py]: This section can be removed and `virtualenv_open` can be added directly to `install_requires` above.

**Comment by asottile** [comment]
looks like we're hitting https://github.com/travis-ci/travis-ci/issues/8363 now

File: tox.ini [hunk 6]
```diff
@@ -107,5 +96,5 @@ exclude_lines = if __name__ == ["']__main__["']:
 addopts = -rsxX
 rsyncdirs = tests tox
 looponfailroots = tox tests
-norecursedirs = .hg .tox
+norecursedirs = .hg {toxworkdir}
```
Comment (asottile) [inline tox.ini]: is this working as intended? I kind of doubt that pytest is going to properly substitute `toxworkdir`

File: tox.ini [hunk 7]
```diff
@@ -48,24 +47,14 @@ commands = python -m flake8 --show-source tox setup.py {posargs}
...
-list_dependencies_command = pip freeze
-
 [testenv:X]
 description = print the positional arguments passed in with echo
 commands = echo {posargs}
 
 [testenv:coverage]
 description = [run locally after tests]: combine coverage data and create reports
 skip_install = True
+changedir={toxworkdir}
```
Comment (asottile) [inline tox.ini]: this is unrelated to the changes here (and it seems breaking windows) -- did you accidentally commit this over another WIP branch?

File: changelog/679.feature.rst [hunk 8]
```diff
@@ -0,0 +1,2 @@
+drop Pyhton ``2.6`` and ``3.3`` support: ``setuptools`` dropped supporting these, and as we depend on it we'll follow
```
Comment (asottile) [inline changelog/679.feature.rst]: Pyhton

File: doc/config-v2.rst [hunk 9]
```diff
@@ -195,7 +195,7 @@ Default settings related to environments names/variants
 tox comes with predefined settings for certain variants, namely:
 
 * ``{easy,pip}`` use easy_install or pip respectively
-* ``{py24,py25,py26,py27,py31,py32,py33,py34,pypy19]`` use the respective
+* ``{py27,py36,py35,py34,pypy19]`` use the respective
```
Comment (asottile) [inline doc/config-v2.rst]: nit: I'd expect these to be in increasing order instead of decreasing

Comment (asottile) [reply doc/config-v2.rst]: same for the rest of the docs -- we should consistently:
- mention pythonx.x, pythony.y in increasing order
- list tox environments in increasing order

I understand this makes your diff less minimal, but it makes the docs read better :)

File: doc/example/result.rst [hunk 10]
```diff
@@ -40,7 +40,7 @@ This will create a json-formatted result file using this schema:
       "platform": "linux2",
       "installpkg": {
         "basename": "tox-1.6.0.dev1.zip",
-        "sha256": "b6982dde5789a167c4c35af0d34ef72176d0575955f5331ad04aee9f23af4326",
+        "sha356": "b6982dde5789a167c4c35af0d34ef72176d0575955f5331ad04aee9f23af4336",
```
Comment (asottile) [inline doc/example/result.rst]: woops, nice sed :D

File: tests/test_config.py [hunk 11]
```diff
@@ -1388,7 +1388,7 @@ def test_factors(self, newconfig):
         conf = newconfig([], inisource)
         configs = conf.envconfigs
         assert [dep.name for dep in configs['a-x'].deps] == \
-            ["dep-all", "dep-a", "dep-x"]
+               ["dep-all", "dep-a", "dep-x"]
```
Comment (asottile) [inline tests/test_config.py]: orthogonal to this change?

File: tox.ini (hunk 7 above)
Comment (gaborbernat) [reply tox.ini]: no accident, will take a look

File: doc/config-v2.rst [hunk 12]
```diff
@@ -253,13 +253,13 @@ Another `tox.ini
 has 233 lines and runs tests against multiple Postgres and Mysql
 engines.  It also performs backend-specific test commands, passing
 different command line options to the test script.  With the new tox-1.X
-we not only can do the same with 32 non-repetive configuration lines but
+we not only can do the same with 35 non-repetive configuration lines but
```
Comment (asottile) [inline doc/config-v2.rst]: I don't think this one was supposed to change

File: doc/example/general.rst [hunk 13]
```diff
@@ -74,7 +74,7 @@ which will make the sphinx tests part of your test run.
 Selecting one or more environments to run tests against
 --------------------------------------------------------
 
-Using the ``-e ENV[,ENV2,...]``  option you explicitly list
+Using the ``-e ENV[,ENV36,...]``  option you explicitly list
```
Comment (asottile) [inline doc/example/general.rst]: this one looks like a mistake too

File: doc/example/nose.rst [hunk 14]
```diff
@@ -25,7 +25,7 @@ and the following ``tox.ini`` content:
 
 you can invoke ``tox`` in the directory where your ``tox.ini`` resides.
 ``tox`` will sdist-package your project create two virtualenv environments
-with the ``python2.6`` and ``python2.5`` interpreters, respectively, and will
+with the ``python3.6`` and ``python2.7`` interpreters, respectively, and will
```
Comment (asottile) [inline doc/example/nose.rst]: ordering here

File: doc/example/unittest.rst [hunk 15]
```diff
@@ -76,8 +76,8 @@ the checkout has a tox.ini_ that looks like this:
 
 mock uses unittest2_ to run the tests. Invoking ``tox`` starts test
 discovery by executing the ``unit2 discover``
-commands on Python 2.4, 2.5, 2.6 and 2.7 respectively.  Against
-Python2.6 and Python2.7 it will additionally run sphinx-mediated
+commands on Python 2.4, 2.5, 3.6 and 2.7 respectively.  Against
```
Comment (asottile) [inline doc/example/unittest.rst]: ordering here, also 2.4 and 2.5

File: doc/example/unittest.rst [hunk 16]
```diff
@@ -50,13 +50,13 @@ the checkout has a tox.ini_ that looks like this:
 .. code-block:: ini
 
     [tox]
-    envlist = py24,py25,py26,py27
+    envlist = py34,py35,py36,py27
```
Comment (asottile) [inline doc/example/unittest.rst]: ordering here

File: doc/example/unittest.rst [hunk 17]
```diff
@@ -19,7 +19,7 @@ and add the following ``tox.ini`` to it:
 .. code-block:: ini
 
     [tox]
-    envlist = py25,py26,py27
+    envlist = py35,py36,py27
```
Comment (asottile) [inline doc/example/unittest.rst]: ordering here

File: doc/index.rst [hunk 18]
```diff
@@ -32,19 +32,19 @@ right next to your ``setup.py`` file::
...
 You can also try generating a ``tox.ini`` file automatically, by running
 ``tox-quickstart`` and then answering a few simple questions.
 
-To sdist-package, install and test your project against Python2.6 and Python2.7, just type::
+To sdist-package, install and test your project against Python2.7 and Python3.6, just type::
 
     tox
 
-and watch things happening (you must have python2.6 and python2.7 installed in your
+and watch things happening (you must have python3.6 and python2.7 installed in your
```
Comment (asottile) [inline doc/index.rst]: ordering here

File: tox/interpreters.py [hunk 19]
```diff
@@ -17,7 +17,7 @@ def __init__(self, hook):
 
     def get_executable(self, envconfig):
         """ return path object to the executable for the given
-        name (e.g. python2.6, python2.7, python etc.)
+        name (e.g. python3.6, python2.7, python etc.)
```
Comment (asottile) [inline tox/interpreters.py]: ordering here

File: doc/example/general.rst (hunk 13 above)
Comment (gaborbernat) [reply doc/example/general.rst]: this one I think is fine :+1:

Comment (asottile) [reply doc/example/general.rst]: I don't or I wouldn't have pointed it out.  This is a listing of environments and yes while most of them are usually python environments this is documenting the command line.
